client/build_code/loaders/sass-interop-loader.js

```diff
@@ -0,0 +1,60 @@
+// Adds ephemeral :export { ... } block to .interop.sass modules during webpack processing.
+// Necessary for css-loader to add exports to generated JS modules, but not worth the headache
+// of managing as hardcoded statements inside .interop.sass modules
+
+const path = require("path");
+
+const _ = require("lodash");
+
+const loader_name = path.basename(__filename); // eslint-disable-line no-undef
+
+// as documented in /client/docs/style-sheets/interop-sass-modules.md, this intentionally only matches on and exports "top level" variables,
+// intenionally ignoring any sass "global" (module level) variables from inside of sass branching syntax blocks (e.g. SASS ifs, loops, etc)
+const sass_variable_pattern = /^\$(.*?):.*/;
+
+const get_top_level_variable_names = (content) =>
+  _.chain(content)
+    .split("\n")
+    .filter((line) => sass_variable_pattern.test(line))
+    .map((export_line) => export_line.replace(sass_variable_pattern, "$1"))
```

Comment:
Just curious what's replacing `$1` for?

Reply:
Regex syntax, you can make references to the matched content by a "capturing group" within your replace pattern. Capturing groups are the portions of patterns within parenthesis (unless flagged as a "non capturing group" by adding `?:` to the start of the group), identified within replacement patterns by `$#` where # are the order the capturing group appears in.

Specifically, `sass_variable_pattern = /^\$(.*?):.*/` so this replace is taking something like `$variableName: ....value` and stripping it down to just `variableName` (the content between the `$` and the `:`).